noir-projects/aztec-nr/aztec/src/utils/conversion/fields_to_bytes.nr

```diff
@@ -0,0 +1,127 @@
+// These functions are used to facilitate the conversion of log plaintext represented as fields into bytes and back.
+//
+// `fields_to_bytes` uses fixed-size arrays since encryption contexts have compile-time size information.
+// `fields_from_bytes` uses BoundedVec for flexibility in unconstrained contexts where sizes are dynamic.
+//
+// Together they provide bidirectional conversion between fields and bytes.
+
+/// Converts an input array of fields into a single array of bytes. Use `fields_from_bytes` to obtain the original
+/// field array.
+/// Each field is converted to a 32-byte big-endian array.
+///
+/// For example, if you have a field array [123, 456], it will be converted to a 64-byte array:
+/// [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,123,  // First field (32 bytes)
+///  0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,200]  // Second field (32 bytes)
+///
+/// Since a field is ~254 bits, you'll end up with a subtle 2-bit "gap" at the big end, every 32 bytes. Be careful
+/// that such a gap doesn't leak information! This could happen if you for example expected the output to be
+/// indistinguishable from random bytes.
+pub fn fields_to_bytes<let N: u32>(fields: [Field; N]) -> [u8; 32 * N] {
+    let mut bytes = [0; 32 * N];
+
+    for i in 0..N {
+        let field_as_bytes: [u8; 32] = fields[i].to_be_bytes();
+
+        for j in 0..32 {
+            bytes[i * 32 + j] = field_as_bytes[j];
+        }
+    }
+
+    bytes
+}
+
+/// Converts an input BoundedVec of bytes into a BoundedVec of fields. Arbitrary byte arrays are not allowed: this
+/// is assumed to be an array obtained via `fields_to_bytes`, i.e. one that actually represents fields. To convert
+/// a byte array into Fields, use `bytes_to_fields`.
+///
+/// The input bytes are chunked into chunks of 32 bytes. Each 32-byte chunk is viewed as big-endian, and is converted
+/// into a Field.
+/// For example, [1, 10, 3, ..., 0] (32 bytes) is encoded as [1 * 256^31 + 10 * 256^30 + 3 * 256^29 + ... + 0]
+/// Note: N must be a multiple of 32 bytes
+pub fn fields_from_bytes<let N: u32>(bytes: BoundedVec<u8, N>) -> BoundedVec<Field, N / 32> {
+    // Assert that input length is a multiple of 32
+    assert(bytes.len() % 32 == 0, "Input length must be a multiple of 32");
+
+    // Since fields are ~254 bits, we must ensure the most significant byte of each 32-byte chunk
+    // contains no more than 6 bits, preventing overflow when converting back to fields.
+    for i in 0..bytes.len() / 32 {
+        let last_byte = bytes.get(i * 32) as Field;
+        last_byte.assert_max_bit_size::<6>();
+    }
```

Comment:
This is either too restrictive or too loose I think. Have tests in which you convert an array with the largest possible field value into bytes and then back.